lib/ruby_lsp/test_helper.rb

```diff
@@ -52,5 +52,37 @@ def with_server(source = nil, uri = Kernel.URI("file:///fake.rb"), stub_no_typec
         server.run_shutdown
       end
     end
+
+    sig { params(server: RubyLsp::Server).returns(RubyLsp::Result) }
+    def pop_result(server)
+      result = server.pop_response
+      result = server.pop_response until result.is_a?(RubyLsp::Result) || result.is_a?(RubyLsp::Error)
+
+      if result.is_a?(RubyLsp::Error)
+        raise "Failed to execute request #{result.message}"
```

Comment:
In `ruby-lsp-rails` this uses `refute_instance_of`, I've changed it to to avoid coupling to the test framework.

Reply:
I think a common practice is to define a custom exception class under `TestHelper` and raise that exception here. This will allow consumers of this helper to assert/refute this case more easily.

Reply:
Yeah, I didn't do so since this isn't implementation code, but I can add.